lib/vrl/stdlib/src/url_util.rs

```diff
@@ -0,0 +1,31 @@
+use url::form_urlencoded::Parse;
+use vrl::prelude::*;
+
+/// Inner kind for a parsed query string.
+pub(super) fn query_inner_kind() -> Collection<Field> {
+    Collection::from_unknown(Kind::bytes().or_array(Collection::any()))
```

Comment:
I believe we know that if it is an array it is going to be an array of bytes, so this should be possible:

```suggestion
    Collection::from_unknown(Kind::bytes().or_array(Collection::from_unknown(Kind::bytes()))
```